src/app/utils/getTopicContents.ts

```diff
@@ -0,0 +1,13 @@
+import { TOPIC_MAP, TopicKey } from "@/app/constants/topics";
+
+function getTopicContents(
+  topics: (typeof TOPIC_MAP)[TopicKey],
+  selectedContents: string[]
+) {
+  return Object.entries(topics)
+    .filter(([key]) => selectedContents.includes(key))
+    .reduce<string[]>((acc, [_, value]) => {
+      return [...acc, ...value];
+    }, []);
+}
```

Comment:
아 이건 저도 회사에서 받았던 리뷰인데요..
reduce + spread operator는 시간 복잡도가 O(n^2)과 같은 안티패턴이라고 하더라구요
그러면 filter + reduce + spread 로서 또다시 수정 전과 같이 array를 세번 순회하게 될 것 같네요.
시간 나시면 [링크](https://www.richsnapp.com/article/2019/06-09-reduce-spread-anti-pattern) 한번 읽어보시길 바랄게요.

```suggestion
function getTopicContents(
  topics: (typeof TOPIC_MAP)[TopicKey],
  selectedContents: string[]
) {
  return Object.entries(topics).reduce<string[]>((acc, [key, value]) => {
    if (selectedContents.includes(key)) {
      acc.push(...value);
    }

    return acc;
  }, []);
}
```

Reply:
넵 감사합니다ㅜㅜ